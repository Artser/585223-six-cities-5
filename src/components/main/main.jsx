import React from "react";
import PropTypes from "prop-types";
import Coord from "../coord/coord";
import {cityType, offerType} from "../../types";
import {connect} from "react-redux";
import CitiesList from "../city-list/city-list";
import withPlaceList from "../../hocs/with-place-list";
import PLaceList from "../place-list/place-list";
import NoOffers from '../no-offers/no-offers';
import Sorting from "../sorting/sorting";
import {getActiveCity, getFilteredOffers} from "../../reducer/reselect";
import {Header} from "../header/header";

export const PlaceListWrapped = withPlaceList(PLaceList);

const Main = (props) => {
  const {offers, activeCity, authorizationStatus, authInfo, handleOfferChange, activeOfferIndex} = props;
  // константы для проверки пустоты ниже
  const isMainPageEmpty = offers.length ? false : true;
  const classNameForMainTag = isMainPageEmpty ? `page__main--index-empty` : ``;
  const classNameForPlaceContainer = isMainPageEmpty ? `cities__places-container--empty` : ``;
  const classNameForSection = isMainPageEmpty ? `cities__no-places` : `cities__places places`;
  const activeOffer = offers.find((offer) => offer.id === activeOfferIndex);


  return <div className="page page--gray page--main">

    <Header
      authorizationStatus={authorizationStatus}
      authInfo={authInfo}
    />

    <main className={`page__main page__main--index ${classNameForMainTag}`}>
      <h1 className="visually-hidden">Cities</h1>
      <CitiesList />
      <div className="cities">
        <div className={`cities__places-container ${classNameForPlaceContainer} container`}>
          <section className={`${classNameForSection}`}>
            {isMainPageEmpty
              ? <NoOffers city={activeCity.name} />
              : <React.Fragment>
                <h2 className="visually-hidden">Places</h2>
                <b className="places__found">{offers.length} places to stay in {activeCity.name}</b>
                <Sorting />


                <PlaceListWrapped
                  offers={offers}
                  onHover={handleOfferChange}
                />
              </React.Fragment>
            }
          </section>
          <div className="cities__right-section">

            {isMainPageEmpty
              ? ``
              : <Coord
                coords={offers.map(({coord}) => coord)}
                activeCity={activeCity}
                activeOffer={activeOffer && activeOffer.coord}
              />}

          </div>
        </div>
      </div>
    </main>
  </div>;
};
Main.propTypes = {
  offers: PropTypes.arrayOf(offerType),
  activeCity: cityType,
  authorizationStatus: PropTypes.string,
  authInfo: PropTypes.object,
  handleOfferChange: PropTypes.func,
  activeOfferIndex: PropTypes.number,

};

const mapStateToProps = (state) => {


  return {
    authorizationStatus: state.authorizationStatus,
    authInfo: state.authInfo,
    offers: getFilteredOffers(state),
    activeCity: getActiveCity(state),
  };
};
export {Main};
export default connect(mapStateToProps)(Main);
