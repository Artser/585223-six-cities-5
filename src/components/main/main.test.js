import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {Main} from './main';
jest.mock(`../city-list/city-list`, () => `CitiesList`);
jest.mock(`../sorting/sorting`, () => `Sorting`);
jest.mock(`../place-list/place-list`, () => `PLaceList`);
jest.mock(`../coord/coord`, () => `Coord`);
jest.mock(`../no-offers/no-offers`, () => `NoOffers`);

describe(`Render Main`, () => {
  const offer = {
    adults: 3,
    avatar: `img/avatar-angelina.jpg`,
    bedrooms: 1,
    city: `Amsterdam`,
    cityId: 4,
    coord: [52.388540000000006, 4.899976],
    description: `A new spacious villa, one floor.`,
    id: 5,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`],
    imgLink: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    isFavorite: true,
    isPremium: false,
    name: `Angelina`,
    price: 134,
    type: `test`,
    rating: 2.2,
    goods: [test],
    threeImages: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`],
    title: `The house among olive`
  };
  const activeCity = {
    name: `Amsterdam`,
    coord: [54, 45],
    id: 2
  };
  const authorizationStatus = `yes`;
  const authInfo = {
    login: `Dusseldorf@mail.ru`,
    password: 1111
  };
  const handleOfferChange = jest.fn();
  const activeOfferIndex = 5;
  const activeOffer = 5;
  it(`Render Main`, () => {

    const tree = renderer
      .create(
          <MemoryRouter>
            <Main
              offers={[offer]}
              activeCity={activeCity}
              authInfo={authInfo}
              handleOfferChange={handleOfferChange}
              activeOfferIndex={activeOfferIndex}
              authorizationStatus={authorizationStatus}
              activeOffer={activeOffer}
            />
          </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
