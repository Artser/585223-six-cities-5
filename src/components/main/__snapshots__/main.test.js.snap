// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main Render Main 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <span
                className="header__nav-link header__nav-link--profile"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <a
                  className="header__login"
                  href="/login"
                  onClick={[Function]}
                >
                  Sign in
                </a>
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CitiesList />
    <div
      className="cities"
    >
      <div
        className="cities__places-container  container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            1
             places to stay in 
            Amsterdam
          </b>
          <Sorting />
          <PLaceList
            activePlaceIndex={-1}
            handleHover={[Function]}
            offers={
              Array [
                Object {
                  "adults": 3,
                  "avatar": "img/avatar-angelina.jpg",
                  "bedrooms": 1,
                  "city": "Amsterdam",
                  "cityId": 4,
                  "coord": Array [
                    52.388540000000006,
                    4.899976,
                  ],
                  "description": "A new spacious villa, one floor.",
                  "goods": Array [
                    [Function],
                  ],
                  "id": 5,
                  "images": Array [
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  ],
                  "imgLink": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  "isFavorite": true,
                  "isPremium": false,
                  "name": "Angelina",
                  "price": 134,
                  "rating": 2.2,
                  "threeImages": Array [
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  ],
                  "title": "The house among olive",
                  "type": "test",
                },
              ]
            }
            onHover={[MockFunction]}
          />
        </section>
        <div
          className="cities__right-section"
        >
          <Coord
            activeCity={
              Object {
                "coord": Array [
                  54,
                  45,
                ],
                "id": 2,
                "name": "Amsterdam",
              }
            }
            activeOffer={
              Array [
                52.388540000000006,
                4.899976,
              ]
            }
            coords={
              Array [
                Array [
                  52.388540000000006,
                  4.899976,
                ],
              ]
            }
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
